### PR TITLE
Lb/organisation views for non support users placements

### PR DESCRIPTION
**Pull request by elceebee**
## Context

As a non-support user, I should be redirected to my organisation on login. If I have more than one organisation, I should be able to choose which organisation I want to view

## Changes proposed in this pull request

- Moved the school detail stuff to shared resources.
- Added views for organisation details.

## Guidance to review

Make sure your different personas have placements organisations attached to them
Login as any non-support persona. If they have one organisation, you will see the organisation details page. If you have more than one, you'll see the organisation list. 
Login as a claims support user and make sure I haven't broken anything by deleting those partials in claims

If the persona has organisations that are not placements (claims or no service at all), you shouldn't see them on your list, and you should not be able to view that organisation.

## Link to Trello card

Don't have access to trello at the moment, so can't like to card

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated or [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

## Screenshots

https://github.com/DFE-Digital/itt-mentor-services/assets/44073106/e183c335-6163-41bf-9e7f-2fcf2218f876

### CODE REVIEW

File: app/controllers/placements/organisations_controller.rb [hunk 1]
```diff
@@ -1,6 +1,22 @@
...
+  def redirect_to_organisation_if_only_one
+    set_memberships
+
+    if @memberships.one?
+      redirect_to placements_organisation_path(organisation)
+    end
+  end
+
+  def set_memberships
+    @memberships = current_user.memberships.filter(&:placements_service).sort_by(&:name)
```
Comment (elceebee) [inline app/controllers/placements/organisations_controller.rb:16]: We'll be able to do this more elegantly when we have done some additional work on scoping (tickets have been written and we're working on those today)

File: app/views/placements/organisations/index.html.erb [hunk 2]
```diff
@@ -2,24 +2,14 @@
   <div class="govuk-grid-row">
```
Comment (JamieCleare2525) [inline app/views/placements/organisations/index.html.erb:2]: This file doesn't set the `<%= content_for :page_title, t(".organisations") %>`

Comment (elceebee) [reply app/views/placements/organisations/index.html.erb:2]: You are right. After all my banging on about this, and then I forgot it!

Comment (elceebee) [reply app/views/placements/organisations/index.html.erb:3]: Done

**Comment by elceebee** [comment]
[Add organisation views to non-support users](https://trello.com/c/MnbOWMcV/145-add-organisation-views-to-non-support-users)

File: app/models/membership.rb [hunk 3]
```diff
@@ -24,4 +24,7 @@ class Membership < ApplicationRecord
   belongs_to :organisation, polymorphic: true
 
   validates :user, uniqueness: { scope: :organisation_id }
+
+  delegate :name, to: :organisation
+  delegate :placements_service, to: :organisation
```
Comment (Nitemaeric) [inline app/models/membership.rb:29]: Tiny one.

```suggestion
  delegate :name, :placements_service, to: :organisation
```

File: app/views/claims/schools/show.html.erb [hunk 4]
```diff
@@ -22,11 +22,11 @@
     <div class="govuk-grid-column-two-thirds">
       <%= render "shared/organisations/organisation_details", organisation: @school %>
       <h2 class="govuk-heading-m govuk-!-margin-top-9"><%= t(".additional_details") %></h2>
-      <%= render "additional_details", school: @school %>
+      <%= render "shared/organisations/school_additional_details", school: @school %>
```
Comment (Nitemaeric) [inline app/views/claims/schools/show.html.erb:25]: Hmm, should schools be nested within the organisations scope?

Just thinking out loud, if it was a shared partial between both providers and schools, I'd expect a `shared/organisations/additional_details` partial with an `organisation` local. As the schools and providers themselves, I'm thinking they might live at `shared/schools` and `shared/providers`, respectively. Although, we'll likely not need `shared/providers` for at least the Private Beta.

```suggestion
      <%= render "shared/schools/additional_details", school: @school %>
```

File: app/views/placements/organisations/index.html.erb [hunk 5]
```diff
@@ -1,25 +1,16 @@
...
-        <h2 class="govuk-heading-l"><%= t(".providers") %></h2>
-        <ul class="govuk-list">
-          <% @providers.each do |provider| %>
-            <li><%= provider.name %></li>
+          <% @memberships.each do |membership| %>
+            <li><%= govuk_link_to(membership.name, placements_organisation_path(membership.organisation)) %></li>
           <% end %>
         </ul>
+      <% else %>
+      <p><%= t(".empty_state") %></p>
```
Comment (Nitemaeric) [inline app/views/placements/organisations/index.html.erb:13]: 👍 I much prefer this key name over `no_{resource_name}`.

**Review by Nitemaeric — COMMENTED** [review]
A few comments. All else looking really good!

**Review by JamieCleare2525 — APPROVED** [review]

**Review by Nitemaeric — APPROVED** [review]
🚀

**Pull request merged**